src/Components/GEOSldas_GridComp/GEOS_LdasGridComp.F90

```diff
@@ -688,11 +709,11 @@ subroutine Initialize(gc, import, export, clock, rc)
     tcinternal%pgrid_g = pert_grid_g
     tcinternal%pgrid_f = pert_grid_f
     tcinternal%pgrid_l = pert_grid_l
-    if(trim(grid_type) == "Cubed-Sphere" ) then
-      tcinternal%grid_g = tile_grid_g
-    else
-      tcinternal%grid_g = tcinternal%pgrid_g
-    endif
+    !if(trim(grid_type) == "Cubed-Sphere" ) then
+    tcinternal%tgrid_g = tile_grid_g
```

Comment:
@weiyuan-jiang: We shouldn't need the if block here, right?

Reply:
Without this if block, we broadcast the tile_grid_g twice when grid_type != "Cubed-Sphere".

Reply:
Yes, but I think it's the only clean way to assign tile_grid_g to tcinternal%tgrid_g here:

https://github.com/GEOS-ESM/GEOSldas/blob/f1922cf84ea2dc59ea55771cbae3862e2c8298ac/src/Components/GEOSldas_GridComp/GEOS_LdasGridComp.F90#L706

Before my changes, this was `tcinternal%[t]grid_g = tcinternal%pgrid_g`, which is somewhat backwards.  
`pgrid_g` is defined in get_pert_grid() based on `tgrid_g`.
While get_pert_grid() in its current version makes `pgrid_g` identical to `tgrid_g` unless we're in cube-sphere tile space, this could change in future revisions.  Then using `tcinternal%grid_g = tcinternal%pgrid_g` would also need to be changed, and that would be very easy to miss.  
I think we should stick to the logic that pgrid is derived from tgrid.  Broadcasting tile_grid_g should only require negligible resources.  It's just a broadcast (not a send/receive) and only a few strings and numbers.